osu.Game/Configuration/SettingSourceAttribute.cs

```diff
@@ -183,5 +184,16 @@ public static ICollection<(SettingSourceAttribute, PropertyInfo)> GetOrderedSett
             => obj.GetSettingsSourceProperties()
                   .OrderBy(attr => attr.Item1)
                   .ToArray();
+
+        private class ModSettingsEnumDropdown<T> : SettingsEnumDropdown<T>
+            where T : struct, Enum
+        {
+            protected override OsuDropdown<T> CreateDropdown() => new ModDropdownControl();
+
+            private class ModDropdownControl : DropdownControl
+            {
+                protected override int MenuMaxHeight => 100;
```

Comment:
Could just directly override `CreateMenu` and do the menu max height change, without adding virtual properties or anything else.

Reply:
I'm not too fussed about having this as a virtual property, or even a plain property. But if that sort of thing is going to happen, it should really happen one level up (in `SettingsDropdown<T>`).

Reply:
The reason I didn't override `CreateMenu` was because I wanted to avoid calling the base method which unnecessarily sets the max height to 200.

The controls in `SettingsDropdown` and `SettingsEnumDropdown` derive from different classes, so including the prop in either one won't make the prop inheritable by the other.

Reply:
> The controls in SettingsDropdown and SettingsEnumDropdown derive from different classes

I'm confused. `SettingsEnumDropdown` derives from `SettingsDropdown`, does it not?

If it's unclear, I was suggesting that the property could be hoisted outside to the dropdown itself rather than be in the inner control class.

Reply:
I ended up overriding `CreateMenu` for simplicity.